File: app/src/main/java/com/example/engineeraitest/presentation/base/BaseActivity.java
package com.example.engineeraitest.presentation.base;

import android.view.View;
import android.view.WindowManager;

import androidx.appcompat.app.AppCompatActivity;

import com.example.engineeraitest.R;

public abstract class BaseActivity extends AppCompatActivity {

    protected void showProgress() {
        findViewById(R.id.progress_bar).setVisibility(View.VISIBLE);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE,
                WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE);
    }

    protected void hideProgress() {
        getWindow().clearFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE);
        findViewById(R.id.progress_bar).setVisibility(View.GONE);
    }

    protected boolean isInProgress() {
        return findViewById(R.id.progress_bar).getVisibility() == View.VISIBLE;
    }
}

File: app/src/main/java/com/example/engineeraitest/presentation/base/BaseActivity.kt
package com.example.engineeraitest.presentation.base

import android.view.View
import android.view.WindowManager
import android.widget.ProgressBar
import androidx.appcompat.app.AppCompatActivity
import com.example.engineeraitest.R

abstract class BaseActivity : AppCompatActivity() {

    fun showProgress() {
        findViewById<ProgressBar>(R.id.progress_bar).visibility = View.VISIBLE
        window.setFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE,
                WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE)
    }

    fun hideProgress() {
        window.clearFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE)
        findViewById<ProgressBar>(R.id.progress_bar).visibility = View.GONE
    }

    fun isInProgress(): Boolean {
        return findViewById<ProgressBar>(R.id.progress_bar).visibility == View.VISIBLE
    }
}